sunpy/map/mapbase.py

```diff
@@ -608,13 +608,23 @@ def wcs(self):
         w2.wcs.crval = u.Quantity([self._reference_longitude, self._reference_latitude])
         w2.wcs.ctype = self.coordinate_system
         w2.wcs.pc = self.rotation_matrix
-        w2.wcs.set_pv(self._pv_values)
         # FITS standard doesn't allow both PC_ij *and* CROTA keywords
         w2.wcs.crota = (0, 0)
         w2.wcs.cunit = self.spatial_units
-        w2.wcs.dateobs = self.date.isot
-        w2.wcs.aux.rsun_ref = self.rsun_meters.to_value(u.m)
+        w2.wcs.set_pv(self._pv_values)
 
+        # If date average exists we should use it.  If it doesn't then we
+        # fallback to dateobs because it can come from many places
+        if self.date_average is not None:
+            w2.wcs.dateavg = self.date_average.isot
+        else:
+            w2.wcs.dateobs = self.date.isot
```

Comment:
Is there a reason that we conditionally set `dateobs`? Can we not always set this and then let the map source/`.date` decide what actually gets propagated up here?
```suggestion
        w2.wcs.dateobs = self.date.isot
        if self.date_average is not None:
            w2.wcs.dateavg = self.date_average.isot
```

Reply:
The reason that @Cadair is doing this – and the reason for my discomfort with this PR – is because the SPICE map doesn't actually have a source subclass.  So, he wants `GenericMap` to put the "correct" time(s) in the WCS despite the `.date` property returning the "incorrect"  time.

Reply:
So would you rather we add a spice source map?